src/farm/library.cairo

```diff
@@ -50,15 +78,274 @@ namespace CarbonableFarmer {
+
+        let (balance) = IERC721.balanceOf(
+            contract_address=carbonable_project_address, owner=contract_address
+        );
+
+        return (balance=balance,);
+    }
+
+    func share{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        address: felt, precision: felt
+    ) -> (share: Uint256) {
+        alloc_locals;
+
+        let (total_deposite) = count(address);
+        let (total_contract) = total_locked();
+
+        let dividend = precision * total_deposite;
+        let dividend_uint256 = Uint256(low=dividend, high=0);
+        let (share, _) = uint256_unsigned_div_rem(dividend_uint256, total_contract);
+        return (share=share,);
+    }
+
+    func registred_owner_of{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        token_id: Uint256
+    ) -> (address: felt) {
+        // [Check] Uint256 compliance
+        with_attr error_message("CarbonableFarmer: token_id is not a valid Uint256") {
+            uint256_check(token_id);
+        }
+        let (address) = registration_.read(token_id);
+        return (address=address,);
+    }
+
+    //
+    // Externals
+    //
+
+    func start_period{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        unlocked_duration: felt, period_duration: felt
+    ) -> (success: felt) {
+        // [Check] Duration inputs validity
+        with_attr error_message("CarbonableFarmer: Invalid period duration") {
+            assert_nn(period_duration);
+        }
+        with_attr error_message("CarbonableFarmer: Invalid locked duration") {
+            assert_le(unlocked_duration, period_duration);
+        }
+
+        // [Effect] Store period information
+        let (current_time) = get_block_timestamp();
+        start_.write(current_time);
+        period_duration_.write(period_duration);
+        unlocked_duration_.write(unlocked_duration);
+
+        return (success=TRUE,);
+    }
+
+    func stop_period{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+        success: felt
+    ) {
+        // [Check] Current period
+        let (current_time) = get_block_timestamp();
+        let (start) = start_.read();
+        let (period_duration) = period_duration_.read();
+        let end = start + period_duration;
+        with_attr error_message("CarbonableFarmer: No current period") {
+            assert_le(current_time, end);
+        }
+
+        // [Effect] Reset period information
+        start_.write(0);
+        period_duration_.write(0);
+        unlocked_duration_.write(0);
+
+        return (success=TRUE,);
+    }
+
+    func deposite{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        token_id: Uint256
+    ) -> (success: felt) {
+        alloc_locals;
+
+        // [Security] Start reetrancy guard
+        ReentrancyGuard._start();
+
+        // [Check] Locked status
+        let (status) = is_locked();
```

Comment:
Do we want to do it in the other way:
- Return TRUE when the period `is locked` 
- Return FALSE when the period is unlocked

Apply for the function.

If we stay like this, can we at least specify this specific behavior in the comment of the function `is_locked` ?

Reply:
I think you misunderstood how error manager works, or I didn't get your point.

In this case, it evaluates if the status if FALSE, if the evaluation fails (meaning status is TRUE) then it throws the error saying it's locked (which is your proposition if I understand well).

Reply:
My bad for this one, you right, i read the error/assert condition in the other way... ^^